src/Attribute/EarlyValidationResolver.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Input\Validation\Attribute;
+
+use Yiisoft\Hydrator\Context;
+use Yiisoft\Hydrator\NotResolvedException;
+use Yiisoft\Hydrator\ParameterAttributeInterface;
+use Yiisoft\Hydrator\ParameterAttributeResolverInterface;
+use Yiisoft\Hydrator\UnexpectedAttributeException;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\ValidatorInterface;
+
+final class EarlyValidationResolver implements ParameterAttributeResolverInterface
+{
+    private ?Result $result = null;
+
+    public function __construct(
+        private readonly ValidatorInterface $validator,
+    ) {
+    }
+
+    public function setResult(?Result $result): void
+    {
+        $this->result = $result;
+    }
+
+    public function getParameterValue(ParameterAttributeInterface $attribute, Context $context): mixed
+    {
+        if (!$attribute instanceof EarlyValidation) {
+            throw new UnexpectedAttributeException(EarlyValidation::class, $attribute);
+        }
+
+        if ($this->result !== null) {
+            $parameterName = $context->getParameter()->getName();
+            $result = $this->validator->validate(
```

Comment:
So if `context->isResolved` is `false`, there is no need to call validator?

Reply:
We still need to run validation. For example, `Required` rule for not resolved value return specific error message "Value is not passed".